CMakeLists.txt

```diff
@@ -95,6 +95,20 @@ if(MSVC)
     set(CMAKE_C_FLAGS_CHECK       "/WX")
 endif(MSVC)
 
+if(MSVC)
+    option(BUILD_WITH_STATIC_RUNTIME "Build the libraries with /MT compiler flag" OFF)
+    if(BUILD_WITH_STATIC_RUNTIME)
+        foreach(flag_var
+            CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE
+            CMAKE_C_FLAGS_MINSIZEREL CMAKE_C_FLAGS_RELWITHDEBINFO
+            CMAKE_C_FLAGS_CHECK)
+            if(${flag_var} MATCHES "/MD")
```

Comment:
Since we do the replace below, which looks the pattern up anyway, is this `if` test necessary? It looks like the `string(REGEX REPLACE...)` alone would work equally well.